src/MultiRegion/cubed_sphere_boundary_conditions.jl

```diff
@@ -0,0 +1,351 @@
+                field_1[region][Nc+1, 1-Hc:0, k]       .= reverse(field_2[region_E][1:Hc, 1, k])'
+                #- E + W Halo for field_2:
+                field_2[region][Nc+1:Nc+Hc, 1:Nc, k]   .=         field_2[region_E][1:Hc, 1:Nc, k]
+                field_2[region][Nc+1:Nc+Hc, Nc+1, k]   .=         field_2[region_N][1:Hc, 1, k]
+                field_2[region][1-Hc:0, 2:Nc+1, k]     .= reverse(field_1[region_W][1:Nc, Nc+1-Hc:Nc, k], dims=1)' * plmn
+                field_2[region][1-Hc:0, 1, k]          .=         field_1[region_S][1, Nc+1-Hc:Nc, k] * plmn
+                #- N + S Halo for field_2:
+                field_2[region][1:Nc, Nc+1:Nc+Hc, k]   .= reverse(field_1[region_N][1:Hc, 1:Nc, k], dims=2)'
+                field_2[region][1:Nc, 1-Hc:0, k]       .=         field_2[region_S][1:Nc, Nc+1-Hc:Nc, k]
+            end
+        elseif iseven(region)
+            #- even face number (2, 4, 6):
+            for k in -Hz+1:Nz+Hz
+                #- E + W Halo for field_1:
+                field_1[region][Nc+1:Nc+Hc, 1:Nc, k]   .= reverse(field_2[region_E][1:Nc, 1:Hc, k], dims=1)'
+                field_1[region][1-Hc:0, 1:Nc, k]       .=         field_1[region_W][Nc+1-Hc:Nc, 1:Nc, k]
+                #- N + S Halo for field_1:
+                field_1[region][1:Nc, Nc+1:Nc+Hc, k]   .=         field_1[region_N][1:Nc, 1:Hc, k]
+                field_1[region][Nc+1, Nc+1:Nc+Hc, k]   .=         field_1[region_E][1, 1:Hc, k]'
+                field_1[region][2:Nc+1, 1-Hc:0, k]     .= reverse(field_2[region_S][Nc+1-Hc:Nc, 1:Nc, k], dims=2)' * plmn
+                field_1[region][1, 1-Hc:0, k]          .=         field_2[region_W][Nc+1-Hc:Nc, 1, k]' * plmn
+                #- E + W Halo for field_2:
+                field_2[region][Nc+1:Nc+Hc, 2:Nc+1, k] .= reverse(field_1[region_E][1:Nc, 1:Hc, k], dims=1)' * plmn
+                field_2[region][Nc+1:Nc+Hc, 1, k]      .= reverse(field_2[region_S][Nc+1-Hc:Nc, 1, k]) * plmn
+                field_2[region][1-Hc:0, 1:Nc, k]       .=         field_2[region_W][Nc+1-Hc:Nc, 1:Nc, k]
+                field_2[region][1-Hc:0, Nc+1, k]       .= reverse(field_1[region_N][1, 1:Hc, k])
+                #- N + S Halo for field_2:
+                field_2[region][1:Nc, Nc+1:Nc+Hc, k]   .=         field_2[region_N][1:Nc, 1:Hc, k]
+                field_2[region][1:Nc, 1-Hc:0, k]       .= reverse(field_1[region_S][Nc+1-Hc:Nc, 1:Nc, k], dims=2)'
+            end
+        end
+    end
+
+    #-- Add one valid field_1, field_2 value next to the corner, that allows to compute vorticity on a wider stencil
+    # (e.g., vort3(0,1) & (1,0)).
+    if Hc > 1
+        for region in 1:6
+            for k in -Hz+1:Nz+Hz
+                #- SW corner:
+                field_1[region][1-Hc:0, 0, k] .= field_2[region][1, 1-Hc:0, k]
+                field_2[region][0, 1-Hc:0, k] .= field_1[region][1-Hc:0, 1, k]'
+            end
+            for k in -Hz+1:Nz+Hz
```

Comment:
@siddharthabishnu is there a reason that the SW corner needs its own loop over k? If not, then I suggest:

```suggestion
            for k in -Hz+1:Nz+Hz
                #- SW corner:
                field_1[region][1-Hc:0, 0, k] .= field_2[region][1, 1-Hc:0, k]
                field_2[region][0, 1-Hc:0, k] .= field_1[region][1-Hc:0, 1, k]'
```

Reply:
@siddharthabishnu I'm committing this but please confirm that it wasn't like that on purpose

Reply:
@navidcy, it wasn't like that on purpose.